---
title: ""
output:
  html_document:
    toc: true
    toc_float: true
    collapsed: false
    number_sections: false
    toc_depth: 1
    #code_folding: hide
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(message=FALSE,warning=FALSE, cache=TRUE)
```

```{r include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(tximport)
library(DESeq2)
library(GenomicFeatures)
library(rtracklayer)
library(genefilter)
library(ggplot2)
library(grDevices)
library(RColorBrewer)
library(gplots)
library(ggrepel)
library(dplyr)
library(pheatmap)
library(plotly)
library(heatmaply)
library(tidyverse)

setwd("~/testGitHub/Testing")
```

# PCA {.tabset}

In order to run DESEQ2 for our purposes, I had to treat R1 and R2 as replicates for each sample. 

## Original
```{r echo=FALSE}
load("KDM4D_dds.RData")
load("KDM4D_rld.RData")
metaData <- read.csv("metadata.csv", row.names = 1)
rv <- rowVars(assay(rld))
select <- order(rv, decreasing=T)[seq_len(min(500,length(rv)))]
pc <- prcomp(t(assay(rld)[select,]))
condition <- metaData$condition
scores <- data.frame(pc$x, condition)
percentage <- round(pc$sdev / sum(pc$sdev) * 100, 2)
percentage <- paste( colnames(scores), "(", paste( as.character(percentage), "%", ")", sep="") )

p <- plot_ly(scores,x=scores$PC1,y=scores$PC2,text=rownames(scores),mode="markers",color=factor(condition),marker=list(size=11))
p <- layout(p,title="", xaxis = list(title = percentage[1]),
            yaxis = list(title = percentage[2]))
p
```

## R1 and R2 treated as replicates **Day 0**
```{r echo=FALSE}
load("Day_0_RLD.RData")
load("Day_0_metaData.RData")

rv <- rowVars(assay(rld))
select <- order(rv, decreasing=T)[seq_len(min(500,length(rv)))]
pc <- prcomp(t(assay(rld)[select,]))
condition <- metaData$condition
scores <- data.frame(pc$x, condition)
percentage <- round(pc$sdev / sum(pc$sdev) * 100, 2)
percentage <- paste( colnames(scores), "(", paste( as.character(percentage), "%", ")", sep="") )

p <- plot_ly(scores,x=scores$PC1,y=scores$PC2,text=rownames(scores),mode="markers",color=factor(condition),marker=list(size=11))
p <- layout(p,title="", xaxis = list(title = percentage[1]),
                 yaxis = list(title = percentage[2]))
p
```

## R1 and R2 treated as replicates **Day 5**
```{r echo=FALSE}
load("Day_5_rld.RData")
load("Day_5_metaData.RData")

rv <- rowVars(assay(rld))
select <- order(rv, decreasing=T)[seq_len(min(500,length(rv)))]
pc <- prcomp(t(assay(rld)[select,]))
condition <- metaData$condition
scores <- data.frame(pc$x, condition)
percentage <- round(pc$sdev / sum(pc$sdev) * 100, 2)
percentage <- paste( colnames(scores), "(", paste( as.character(percentage), "%", ")", sep="") )

p <- plot_ly(scores,x=scores$PC1,y=scores$PC2,text=rownames(scores),mode="markers",color=factor(condition),marker=list(size=11))
p <- layout(p,title="", xaxis = list(title = percentage[1]),
                 yaxis = list(title = percentage[2]))
p
```

# Overlapping Genes {.tabset}

Criteria for DGE: 2-fold change in gene expression compared to control and p.adj < 0.05

## Day 0
<table>
  <tr>
    <td align="middle">**Upregulated Genes**</td>
     <td align="middle">**Downregulated Genes**</td>
  </tr>
  <tr>
    <td valign="top"><img src="C:/users/Reinnier Padilla/Documents/testGitHub/Testing/upregulated_venn_diagramm_Day0.png"></td>
    <td valign="top"><img src="C:/users/Reinnier Padilla/Documents/testGitHub/Testing/downregulated_venn_diagramm_Day0.png"></td>
  </tr>
 </table>
 
## Day 5
<table>
  <tr>
    <td align="middle">**Upregulated Genes**</td>
     <td align="middle">**Downregulated Genes**</td>
  </tr>
  <tr>
    <td valign="top"><img src="C:/users/Reinnier Padilla/Documents/testGitHub/Testing/upregulated_venn_diagramm_Day5.png"></td>
    <td valign="top"><img src="C:/users/Reinnier Padilla/Documents/testGitHub/Testing/downregulated_venn_diagramm_Day5.png"></td>
  </tr>
 </table>

